### PR TITLE
VX-139: /oauth2/token generate access_token

### PR DESCRIPTION
**Pull request by yarencheng**
Depends on PR: #7 #12 #9

### CODE REVIEW

File: src/httphandler/token_test.go [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+	"github.com/linkernetworks/oauth/src/mocks"
+
+	"github.com/RangelReale/osin"
+	"github.com/gin-gonic/gin"
+	"github.com/linkernetworks/oauth/src/config"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/mock"
+)
+
+func Test_Token(t *testing.T) {
```
Comment (c9s) [inline src/httphandler/token_test.go]: Please don’t use underscore in the name

**Review by c9s — CHANGES_REQUESTED** [review]

File: src/mocks/osin_storage_mock.go [hunk 2]
```diff
@@ -0,0 +1,94 @@
+package mocks
```
Comment (c9s) [inline src/mocks/osin_storage_mock.go]: package name should be singular

File: src/mocks/osin_storage_mock.go [hunk 3]
```diff
@@ -0,0 +1,94 @@
...
+
+import (
+	"github.com/RangelReale/osin"
+	"github.com/stretchr/testify/mock"
+)
+
+type OsinStorageMock struct {
```
Comment (c9s) [inline src/mocks/osin_storage_mock.go]: you are already in the osin package, you don't have to prefix it.

File: src/mocks/osin_storage_mock.go [hunk 4]
```diff
@@ -0,0 +1,94 @@
...
+	mock.Mock
+}
+
+func (m *OsinStorageMock) Clone() osin.Storage {
```
Comment (c9s) [inline src/mocks/osin_storage_mock.go]: we don't need Mock, if the memory storage is faster enough for testing

**Review by c9s — CHANGES_REQUESTED** [review]

**Comment by yarencheng** [comment]
Done. `src/mocks/` is removed @c9s

**Review by c9s — APPROVED** [review]

**Pull request merged**